bolt-sidecar/src/state/execution.rs

```diff
@@ -77,14 +81,12 @@ pub struct ExecutionState<C> {
 }
 
 impl<C: StateFetcher> ExecutionState<C> {
-    /// Creates a new state with the given client. Initializes the `head` and `basefee` fields
-    /// with the current head and basefee.
-    pub async fn new(client: C, head: ChainHead) -> Result<Self, StateError> {
-        let basefee = client.get_basefee(Some(head.block())).await?;
-
+    /// Creates a new state with the given client. Does not initialize the state
+    /// with any head block number or state. This should be done with [`update_head`].
+    pub async fn new(client: C) -> Result<Self, TransportError> {
         Ok(Self {
-            head,
-            basefee,
+            basefee: 0,
```

Comment:
Why not initialize it? If we don't do this the commitment requests before the first update_head might fail i think

Reply:
restored this!